### PR TITLE
Upgrade sample to 8.0-preview6

### PR DESCRIPTION
**Pull request by mitchdenny**
A few minimal changes to bring up to preview 6 standard plus a few fixes.

### CODE REVIEW

**Comment by mitchdenny** [comment]
@PhilStollery you created this sample. Did you have any concerns with this upgrade? If I don't hear back I'll merge it.

**Comment by mitchdenny** [comment]
@bradygaster  @IEvangelist  I might need your help to merge this one?

File: end/eShop.AppHost/Program.cs [hunk 1]
```diff
@@ -17,32 +15,30 @@
...
     .WithReference(basketApi)
-    .WithReference(idp)
-    // Force HTTPS profile for web app (required for OIDC operations)
-    .WithLaunchProfile("https");
+    .WithReference(idp);
 
 // Inject the project URLs for Keycloak realm configuration
-idp.WithEnvironment("WEBAPP_HTTP", () => webApp.GetEndpoint("http").Value);
-idp.WithEnvironment("WEBAPP_HTTPS", () => webApp.GetEndpoint("https").Value);
+idp.WithEnvironment("WEBAPP_HTTP", $"{webApp.GetEndpoint("http")}");
```
Comment (IEvangelist) [inline end/eShop.AppHost/Program.cs:38]: Why not pass the `EndpointReference` instead of `.ToString` here?

**Review by IEvangelist — APPROVED** [review]

File: end/eShop.AppHost/Program.cs (hunk 1 above)
Comment (nickna) [reply end/eShop.AppHost/Program.cs:38]: I believe `Value` was removed in favor of string `Url`

Comment (mitchdenny) [reply end/eShop.AppHost/Program.cs:38]: Thanks for calling this out @IEvangelist ... the question helped me find what I think might be a bug. As it happens the bug is helpful here:

https://github.com/dotnet/aspire/issues/3937

I think we can merge this as is but we need to think about how we want to handle this on the Aspire side and we _might_ need to revisit this sample.

**Pull request merged**